d2/src/test/java/com/linkedin/d2/jmx/RelativeLoadBalancerStrategyJmxTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.d2.jmx;
+
+import com.linkedin.d2.balancer.clients.TrackerClient;
+import com.linkedin.d2.balancer.strategies.relative.PartitionState;
+import com.linkedin.d2.balancer.strategies.relative.RelativeLoadBalancerStrategy;
+import com.linkedin.d2.balancer.strategies.relative.TrackerClientMockHelper;
+import com.linkedin.d2.balancer.strategies.relative.TrackerClientState;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.mockito.Mockito;
+import org.testng.annotations.Test;
+
+import static org.easymock.EasyMock.anyInt;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+
+public class RelativeLoadBalancerStrategyJmxTest {
+  private RelativeLoadBalancerStrategyJmx mockRelativeLoadBalancerStrategyJmx(List<TrackerClient> trackerClients)
+  {
+    Map<TrackerClient, TrackerClientState> trackerClientsMap = new HashMap<>();
+    for (TrackerClient trackerClient : trackerClients)
+    {
+      trackerClientsMap.put(trackerClient, new TrackerClientState(1, 1));
+    }
+
+    RelativeLoadBalancerStrategy strategy = Mockito.mock(RelativeLoadBalancerStrategy.class);
+    PartitionState state = Mockito.mock(PartitionState.class);
+    Mockito.when(state.getTrackerClientStateMap()).thenReturn(trackerClientsMap);
+    Mockito.when(strategy.getPartitionState(anyInt())).thenReturn(state);
+
+    return new RelativeLoadBalancerStrategyJmx(strategy);
+  }
+
+  @Test
+  public void testLatencyDeviation()
+  {
+    List<TrackerClient> trackerClientsEqual = TrackerClientMockHelper.mockTrackerClients(2,
+        Arrays.asList(20, 20), Arrays.asList(10, 10), Arrays.asList(200L, 200L), Arrays.asList(100L, 100L), Arrays.asList(0, 0));
+
+    RelativeLoadBalancerStrategyJmx jmx = mockRelativeLoadBalancerStrategyJmx(trackerClientsEqual);
+    assertEquals(jmx.getLatencyStandardDeviation(), 0.0);
+    assertEquals(jmx.getLatencyMeanAbsoluteDeviation(), 0.0);
+    assertEquals(jmx.getAboveAverageLatencyStandardDeviation(), 0.0);
+
+    List<TrackerClient> trackerClientsDiverse1 = TrackerClientMockHelper.mockTrackerClients(3,
+        Arrays.asList(20, 20, 20), Arrays.asList(10, 10, 10),
+        Arrays.asList(100L, 150L, 200L), Arrays.asList(50L, 75L, 100L), Arrays.asList(0, 0, 0));
```

Comment:
Let's also have a unit test where the call count and latency for some of the hosts are 0.

Reply:
Added in the latest commit.